tests/4.5/taskloop/test_taskloop_collapse.c

```diff
@@ -0,0 +1,58 @@
+//------------------------------------------- test_taskloop_collapse.c --------------------------------------//
+//
+// OpenMP API Version 4.5 September 2015
+//
+// This test checks the 'taskloop' directive with the 'collapse' clause specified.
+// The 'taskloop' construct parallelize loops with independent iterations by creating tasks. 
+// It allows for efficient parallel execution of loop iterations by distributing them among multiple threads. 
+// The 'collapse' clause ensures that multiple nested loops into a single loop.      
+//----------------------------------------------------------------------------------------------------------//
+
+#include <stdio.h>
+#include <omp.h>
+#include "ompvv.h"
+
+#define NUM_THREADS 100
+#define NUM_COLLAPSE 2
+
+int test_taskloop_collapse() {
+
+  int errors = 0;
+
+  int WEEK = 1, DAYS_IN_WEEK = 7, sum_with_openmp = 0, sum_without_openmp = 0;
+
+   //get valid sum without openmp
+   for(int i = 1; i <= (WEEK*DAYS_IN_WEEK); i++) {
+     sum_without_openmp = sum_without_openmp + i;
+   }
+
+   #pragma omp parallel num_threads(NUM_THREADS)
+   {
+      #pragma omp single
+      {
+	#pragma omp taskloop collapse(NUM_COLLAPSE) 
+        for(int i = 1; i <= WEEK; i++) 
+	{
+	    for(int j = 1; j <= DAYS_IN_WEEK; j++) 
+	    {
+		    #pragma omp atomic 
+		    sum_with_openmp = sum_with_openmp + (i*j);
```

Comment:
if WEEK is not 1, let's say 2, this operation doesn't give the correct value. And in order to test collapse better, WEEK should be greater than 1

Reply:
https://github.com/SOLLVE/sollve_vv/pull/782/commits/24882279d1b6bd4bd0a2f3639ee75976d06dbf93